source/pool/contracts/Pool.sol

```diff
@@ -128,34 +128,53 @@ contract Pool is IPool, Ownable2Step {
   /**
    * @notice Set claims from previous pool contract
    * @dev Only owner
-   * @param _root bytes32
+   * @param _groupId bytes32
    * @param _accounts address[]
    */
   function setClaimed(
-    bytes32 _root,
+    bytes32 _groupId,
     address[] memory _accounts
   ) external override multiAdmin {
-    if (roots[_root] == false) {
-      roots[_root] = true;
-    }
+    // if (roots[rootsByGroupId[_groupId]] == false) {
+    //   roots[rootsByGroupId[_groupId]] = true;
+    // }
     for (uint256 i = 0; i < _accounts.length; i++) {
```

Comment:
You will save a lot of gas here if you extract `bytes32 root = rootsByGroupId[_group]` outside this for loop. As it is you're reading from storage on every iteration for something that isn't going to change.

Even without the loop, you're reading this from storage twice, once here, and once at the `Emit` line lower down. 

**NOTE** if you agree that `claimed` should be keyed by the `groupId` then you will need to change this anyway so the comment may not be relevant.

Reply:
indeed thanks for the catch